Add version commands for more languages/executables (#2541)
DOMjudge/domjudge: doc/manual/install-language.rst

Appendix: Installing the example languages
==========================================

DOMjudge default ships with some default languages with a default configuration.
As you might setup contests with those languages we provide how those languages were
installed in the past as guideline.

Most of those can be found in the table below as there is a package available to install

> inside the judging chroot. Given that you can install your own chroot we only provide the

Reviewer: Perhaps mention how it can be installed in the chroot?

Author: Yes, will do.

packages for Ubuntu as that is the most used at the moment of writing.

.. list-table:: Packages for languages
   :header-rows: 1

   * - Language
     - Ubuntu package
     - Remarks
   * - Ada
     - `gnat`
     -
   * - AWK
     - `mawk`/`gawk`
     - `mawk` is default installed
   * - Bash
     - `bash`
     - Default installed in the chroot
   * - C
     - `gcc`
     - Default installed in the chroot
   * - C++
     - `g++`
     - Default installed in the chroot
   * - C#
     - `mono-mcs`
     -
   * - Fortran
     - `gfortran`
     -
   * - Haskell
     - `ghc`

>      - Currently not working

Author: No, there are a couple more where I still need to do some extra work.

The original idea of the PR was to get the version commands, finding out that the language is not usable with those instructions is part 2 IMO.

>      - Currently not working

Author: I documented the why here: https://github.com/DOMjudge/domjudge/issues/2551

installed in the past as guideline.

Most of those can be found in the table below as there is a package available to install
inside the judging chroot. Given that you can install your own chroot we only provide the
packages for Ubuntu as that is the most used at the moment of writing.

.. list-table:: Packages for languages
   :header-rows: 1

   * - Language
     - Ubuntu package
     - Remarks
   * - Ada
     - `gnat`
     -
   * - AWK
     - `mawk`/`gawk`
     - `mawk` is default installed
   * - Bash
     - `bash`
     - Default installed in the chroot
   * - C
     - `gcc`
     - Default installed in the chroot
   * - C++
     - `g++`
     - Default installed in the chroot
   * - C#
     - `mono-mcs`
     -
   * - Fortran
     - `gfortran`
     -
   * - Haskell
     - `ghc`
     - Currently not working
   * - Java
     - `default-jdk-headless`
     - Default installed in the chroot
   * - Javascript
     - `nodejs`
     -
   * - Kotlin
     - `kotlin`
     -
   * - Lua
     - `lua5.4`
     - Ubuntu does not ship a generic meta package (yet).
   * - Pascal
     - `fp-compiler`
     -
   * - Perl
     - `perl-base`
     - Default installed in the chroot

>    * - POSIX

Reviewer 2: ```suggestion
   * - POSIX shell
```

     - `dash`
     - Default installed in the chroot
   * - Prolog
     - `swi-prolog-core`
     -
   * - Python3
     - `pypy3`/`python3`
     - Default installed in the chroot.

>        DOMjudge assumes `pypy3` as it runs generally faster.
>        Consider the `PyPy3 PPA`_ if you need the latest python3 features. PyPy3 does not have 100%
>        compatibility with all non-standard libraries in case this is needed you should reconsider the default
>        CPython implementation.

Reviewer 2: ```suggestion
       DOMjudge assumes `pypy3` as it runs faster in general.
       Consider the `PyPy3 PPA`_ if you need the latest python3 features. PyPy3 does not have 100%
       compatibility with all non-standard libraries. In case this is needed you should reconsider the default
       CPython implementation.
```

Appendix: Installing the example languages
==========================================

DOMjudge ships with some default languages with a default configuration.

> As you might setup contests with those languages we provide how those languages were

Reviewer 2: Words like "setup" are written together when used as noun, but separate when used as verbs.
```suggestion
As you might set up contests with those languages we provide how those languages were
```

Reviewer 2: The text doesn't read so smoothly, but just merge after this change. I'll see if I can go through the whole documentation some time and improve things.
